modules/core/shared/src/main/scala/io/circe/Codec.scala

```diff
@@ -0,0 +1,38 @@
+package io.circe
+
+/**
+  *
+  * Convenience typeclass to be used when you need both an [[io.circe.Encoder]] and
+  * [[io.circe.Decoder]] for your type.
+  *
+  * @author Lorand Szakacs, lsz@lorandszakacs.com
+  * @since 21 Dec 2017
+  *
+  */
+trait Codec[A] extends Encoder[A] with Decoder[A]
+
+object Codec {
+  def apply[A](implicit instance: Codec[A]): Codec[A] = instance
+
+  def instance[A](encode: Encoder[A], decode: Decoder[A]): Codec[A] =
```

Comment:
WDYT about making this curried? It'll probably be easier to pass anonymous functions:

```scala
case class Foo(value: String_

//more or less
Codec.instance[Foo] {
  _.value.asJson
} {
  case "foo" => Foo("Foo")
  //...
}
```